### DISCUSSION TITLE
Feature Request: Option to Hide Tabs When Not in Use #2803

### DISCUSSION THREAD

**Discussion by FormalSnake** [ghostty-org/ghostty#2804] [Feature Requests, Ideas]
## Feature Request

I would like to request a new feature that allows users to hide tabs when they are not actively being used or interacted with.

### Description

Currently, tabs in Ghostty are always visible when enabled. While this is useful for many users, some prefer a cleaner interface with less visual clutter. An option to automatically hide tabs when they're not being used would be beneficial for these users.

### Proposed Functionality

- Add a new configuration option, perhaps called `hide-inactive-tabs` or `auto-hide-tabs`
- When enabled, this option would cause the tab bar to be hidden by default
- The tab bar would appear when:
  - A tab is open
- The tab bar should be hidden (not an empty space like it is now)

### Use Cases

1. Users who prefer a minimalist interface
2. Users who primarily use keyboard shortcuts for navigation and rarely need to see the tab bar
3. Users who want to maximize screen space for terminal content

### Possible Implementation

This could potentially be implemented as an extension of the existing `gtk-tabs-location` option, or as a separate boolean configuration.

### Additional Considerations

- The hiding/showing animation should be smooth to avoid jarring transitions
- There should be a keyboard shortcut to toggle the tab bar visibility manually
- The feature should respect the user's choice of tab location (top, bottom, etc.)

Thank you for considering this feature request. Let me know if you need any additional information or clarification.
As an example, iTerm2 implements this exact feature.

**Comment by rhodes-b** [comment]
does `toggle_window_decorations` solve your usecase?

~~it includes the tab-bar (and titlebar if you are using client side decor)~~ edit looks like it does not include the tab bar
linux hides the tab bar when there are no tabs anyways

**Reply by feakuru** [reply]
I think what they meant is "only show tab bar when hovered with a mouse or when tabs are actively switched between", not just "hide the tab bar forever"

**Reply by rhodes-b** [reply]
> There should be a keyboard shortcut to toggle the tab bar visibility manually

This is listed as a consideration which can solve this usecase

> only show tab bar when hovered with a mouse 

if its entirely hidden how can you show a widget on mouse hover? (there is nothing to hover)

another solution in linux is `toggle_tab_overview` which opens a widget with all the active tabs so again if you want to interact with them you don't need a tab bar.  This works even when tabs are hidden

No that is not a solution for show tab bar on swapping tabs or making a new tab im simply listing options people may not know exist that could already solve this workflow

**Reply by feakuru** [reply]
thanks, that option turned out to be the one for me personally :)

> nothing to hover

apps that do this usually show the tab bar when your mouse moves to the top of the window - I suspect it's going to be a pain to implement though. rereading OP's post, I'm actually not sure they meant that after all, so apologies for this interjection
